pkg/network/probe_handler_test.go

```diff
@@ -102,39 +102,47 @@ func TestProbeHandlerSuccessfulProbe(t *testing.T) {
 	}
 }
 
-func BenchmarkProbeHandler(b *testing.B) {
-	var h http.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("Inner Body"))
-	})
+func BenchmarkProbeHandlerNoProbeHeader(b *testing.B) {
+	var h http.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
 	h = NewProbeHandler(h)
-	ts := httptest.NewServer(h)
-	defer ts.Close()
-	options := []interface{}{
-		prober.ExpectsStatusCodes([]int{http.StatusOK}),
-	}
+	resp := httptest.NewRecorder()
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
 
-	b.Run("sequential", func(b *testing.B) {
+	b.Run("sequential-no-header", func(b *testing.B) {
 		for j := 0; j < b.N; j++ {
-			got, err := prober.Do(context.Background(), network.AutoTransport, ts.URL, options...)
-			if err != nil {
-				b.Errorf("Do = %v", err)
-			}
-			if !got {
-				b.Errorf("Unexpected probe result: got: %t, want: true", got)
+			h.ServeHTTP(resp, req)
+		}
+	})
+
+	b.Run("parallel-no-header", func(b *testing.B) {
+		b.RunParallel(func(pb *testing.PB) {
+			for pb.Next() {
+				h.ServeHTTP(resp, req)
 			}
+		})
+	})
+}
+
+func BenchmarkProbeHandlerWithProbeHeader(b *testing.B) {
+	resp := httptest.NewRecorder()
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
+	req.Header.Set(ProbeHeaderName, ProbeHeaderValue)
+	// This will cause 1 alloc
```

Comment:
Hm, not quite true. The handler itself causes the alloc. I would just drop this comment at this point 🤷‍♂ .